apps/studio/components/interfaces/Integrations/Vault/Secrets/EditSecretModal.tsx

```diff
@@ -1,158 +1,214 @@
 import { isEmpty } from 'lodash'
-import { useEffect, useState } from 'react'
+import { useState, useEffect } from 'react'
+import { type SubmitHandler, useForm } from 'react-hook-form'
+import { zodResolver } from '@hookform/resolvers/zod'
+import { z } from 'zod'
 import { toast } from 'sonner'
-
 import { useProjectContext } from 'components/layouts/ProjectLayout/ProjectContext'
 import { GenericSkeletonLoader } from 'components/ui/ShimmeringLoader'
 import { useVaultSecretDecryptedValueQuery } from 'data/vault/vault-secret-decrypted-value-query'
 import { useVaultSecretUpdateMutation } from 'data/vault/vault-secret-update-mutation'
 import type { VaultSecret } from 'types'
-import { Button, Form, Input, Modal } from 'ui'
+import {
+  Button,
+  Dialog,
+  DialogContent,
+  DialogFooter,
+  DialogHeader,
+  DialogSection,
+  DialogSectionSeparator,
+  DialogTitle,
+  Form_Shadcn_,
+  FormControl_Shadcn_,
+  FormField_Shadcn_,
+  Input_Shadcn_,
+} from 'ui'
 import { EyeOff, Eye } from 'lucide-react'
+import { FormItemLayout } from 'ui-patterns/form/FormItemLayout/FormItemLayout'
 
 interface EditSecretModalProps {
-  selectedSecret: VaultSecret | undefined
+  secret: VaultSecret | undefined
   onClose: () => void
 }
 
-const EditSecretModal = ({ selectedSecret, onClose }: EditSecretModalProps) => {
+const SecretSchema = z.object({
+  name: z.string().min(1, 'Please provide a name for your secret'),
+  description: z.string().optional(),
+  secret: z.string().min(1, 'Please enter your secret value'),
+})
+
+const EditSecretModal = ({ secret, onClose }: EditSecretModalProps) => {
   const [showSecretValue, setShowSecretValue] = useState(false)
   const { project } = useProjectContext()
-
-  const { mutateAsync: updateSecret } = useVaultSecretUpdateMutation()
-
-  let INITIAL_VALUES = {
-    name: selectedSecret?.name ?? '',
-    description: selectedSecret?.description ?? '',
-    secret: selectedSecret?.decryptedSecret ?? '',
-  }
+  const formId = 'edit-vault-secret-form'
+  const { data, isLoading: isLoadingSecretValue } = useVaultSecretDecryptedValueQuery(
+    {
+      projectRef: project?.ref!,
+      id: secret?.id!,
+      connectionString: project?.connectionString,
+    },
+    { enabled: secret !== undefined && !!(project?.ref && secret?.id) }
+  )
+  const form = useForm<z.infer<typeof SecretSchema>>({
+    resolver: zodResolver(SecretSchema),
+    defaultValues: {
+      name: secret?.name ?? '',
+      description: secret?.description ?? '',
+      secret: secret?.decryptedSecret ?? data ?? '',
+    },
+    values: {
+      name: secret?.name ?? '',
+      description: secret?.description ?? '',
+      secret: secret?.decryptedSecret ?? data ?? '',
+    },
```

Comment:
this might seem like unnecessary duplication but I found that this was what was needed to fill in the values that are loaded in asynchronously from `useVaultSecretDecryptedValueQuery`. it's better than the odd hook inside of render function that was here previously